src/parser/flow_lexer.ml

```diff
@@ -724,65 +767,86 @@ let token (env: Lex_env.t) lexbuf : result =
       is_tail
     ))
 
-  | binnumber, (letter | '2'..'9'), Star alphanumeric ->
+  | binbigint ->
+    Token (env, T_BIGINT { kind = BINARY; raw = lexeme lexbuf })
+  | (binnumber | binbigint), (letter | '2'..'9'), Star alphanumeric ->
```

Comment:
And actually this is incorrect. This needs to be split into two cases:

1. `| binnumber, (letter | '2'..'9'), Star alphanumeric ->`
2. `| binbigint, Plus alphanumeric ->`

Why? It's because adding a `0` or `1` to the end of a `binnumber` is ok (e.g. `0x111`) but adding `0` or `1` to the end of a `binbigint` is NOT ok (e.g. `0x11n1`)

Reply:
If I do something like this 
```ocaml
| binbigint, Plus alphanumeric ->
  recover env lexbuf ~f:(fun _ _ -> match%sedlex lexbuf with
  | _ -> failwith "unreachable bigint"
  )
| binnumber, (letter | '2'..'9'), Star alphanumeric ->
  recover env lexbuf ~f:(fun _ _ -> match%sedlex lexbuf with
  | _ -> failwith "unreachable number"
  )
```
It always fails with "unreachable number" on `0b101011101n` literal. Why first matcher doesn't work?

Reply:
In your example, `0b101011101n` does NOT match the first pattern. Remember, `binbigint` includes the `n`. So while `0b101011101nx` would match the first pattern, `0b101011101n` would not.

I would instead recommend writing 

```ocaml
| binbigint, Plus alphanumeric ->
  recover env lexbuf ~f:(fun _ _ -> match%sedlex lexbuf with
  | _ -> failwith "unreachable bigint"
  )
| binbigint ->
   Token (env, T_BIGINT { kind = BINARY; raw = lexeme lexbuf })
| binnumber, (letter | '2'..'9'), Star alphanumeric ->
  recover env lexbuf ~f:(fun _ _ -> match%sedlex lexbuf with
  | _ -> failwith "unreachable number"
  )
| binnumber ->
   Token (env, T_NUMBER { kind = BINARY; raw = lexeme lexbuf })
```

* `0b101nx` would match the first pattern
* `0b101n` would match the second pattern
* `0b101x` would match the third pattern
* `0b101` would match the fourth pattern